### PR TITLE
[ticket/12441] Correct database size detection in MariaDB 10

### PR DESCRIPTION
**Pull request by lavigor**
Code changes were taken from [this post](https://www.phpbb.com/community/viewtopic.php?p=13611421#p13611421).

PHPBB3-12441

### CODE REVIEW

**Comment by lavigor** [comment]
Should I also create PR for 3.2?

File: phpBB/includes/functions_admin.php [hunk 1]
```diff
@@ -3101,9 +3101,9 @@ function get_database_size()
 			{
 				$version = $row['mysql_version'];
 
-				if (preg_match('#(3\.23|[45]\.)#', $version))
+				if (preg_match('#(3\.23|[45]\.|10\..+Maria)#', $version))
```
Comment (marc1706) [inline phpBB/includes/functions_admin.php]: I'd suggest a more restrictive regex:
```#(3\.23|[45]\.|10\.[0-9]\.[0-9]{1,2}-+Maria)#````

**Comment by marc1706** [comment]
We should be able to merge this into master so you don't need to submit a PR for the master branch.

**Comment by lavigor** [comment]
@marc1706 updated to your regex, but I haven't found a mess on GitHub.
Should this code be updated again if MariaDB gets the 10.10.x update or MySQL is updated?
Or maybe it can be changed to set the minimum requirement?

**Comment by marc1706** [comment]
With github messing around I meant that github changes the regex if I post it here.
This regex should work for a rather extended timeframe. Both MySQL 6.x and MariaDB 10.10.x are a long time away. This regex is to ensure that the specified command properly works. Having this more or less restrictive is intended.

**Comment by lavigor** [comment]
OK. Thank you for the explanation.

**Pull request merged**